commands/deploy.go

```diff
@@ -146,7 +146,8 @@ func runDeployCommand(args []string, image string, fprocess string, functionName
 	}
 
 	var services stack.Services
-	if len(yamlFile) > 0 {
+
+	if len(yamlFile) > 0 && len(functionName) == 0 {
```

Comment:
I would rather not affect the build command, the way to filter when building is to pass `--filter FN_NAME`